### PR TITLE
yashim/perf: add support for AVIF image format. Gatsby3 + Webpack5 upgrade.

### PR DESCRIPTION
**Pull request by yashim-deriv**
Changes:
- Upgraded to `Gatsby`,  `Webpack` to latest versions.
- Most broken dependencies + plugins are fixed. (Refer below for list of broken/unsupported Gatsby Plugins)
- Migrated from `gatsby-image` to `gatsby-image-plugin`
- Updated all graphql query to use new format.
- Updated old image file calls to use `query-image` plugin
- Added support for `AVIF` format with fallbacks to `webp` and `jpeg` (Only on large images to maintain fast build times)

Side Notes:
- ~`<BackgroundImage>` component uses legacy fluid graphql query. No replacement yet in `gatsby-image-plugin` **Added a property mapper to add support for `gatsby-background-image`** (Will remove once GBI supports gatsby-plugin-image natively)~ **Gatsby Background Image is FULLY SUPPPORTED** 🥳 🎉 

Broken Important Packages:
- `gatsby-plugin-stylelint` **FULLY Supported**

Unsupported Packages (Warnings):
- ~`gatsby-plugin-webpack-bundle-analyser-v2`~ **REMOVED (Unsupported)**
- `gatsby-plugin-manifest` (Warnings but still functional)

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [x] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [x] Improve performance
-   [ ] Style only
-   [x] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/custom/_dhero.js [hunk 1]
```diff
@@ -202,10 +202,12 @@ const query = graphql`
...
         dtrader: file(relativePath: { eq: "dtrader_trade.png" }) {
             childImageSharp {
-                fluid(maxWidth: 1024, srcSetBreakpoints: [1440]) {
-                    ...GatsbyImageSharpFluid_withWebp_noBase64
-                    originalName
-                }
+                gatsbyImageData(
+                    quality: 70
+                    breakpoints: [360, 992, 1440]
+                    placeholder: NONE
```
Comment (sean-binary) [inline src/components/custom/_dhero.js:208]: Do we want dominant colour for this placeholder?

File: src/components/svgs/signals.js [hunk 2]
```diff
@@ -11,23 +11,15 @@ const Signals = ({ dynamic_id }) => (
...
+                transform="translate(-724 -216) translate(700 66) translate(8 142) translate(16 8)"
+            />
```
Comment (sean-binary) [inline src/components/svgs/signals.js:22]: Could you minify this too? 🙏

File: src/components/layout/nav.js [hunk 3]
```diff
@@ -602,14 +619,12 @@ export const Nav = ({ base, is_ppc_redirect, is_ppc }) => {
...
-                </Show.Desktop>
+                <NavDesktop
+                    base={base}
+                    is_ppc={is_ppc}
+                    is_ppc_redirect={is_ppc_redirect}
+                    is_logged_in={is_logged_in}
+                />
                 <Show.Mobile min_width="bp1060">
                     <NavMobile is_ppc={is_ppc} is_logged_in={is_logged_in} />
                 </Show.Mobile>
```
Comment (sean-binary) [inline src/components/layout/nav.js:630]: Do you want a `NavMobileWrapper` as well for consistency?

File: src/pages/home/_what-our-clients-say.js [hunk 4]
```diff
@@ -238,17 +236,15 @@ const WhatOurClientsSay = () => {
...
-                                quote={trader.quote}
-                                name={trader.name}
-                                location={trader.location}
-                                img_path={trader.img_path}
-                                img_alt={trader.name + localize(" - Deriv's Client")}
-                            />
-                        </div>
+                    {our_client_slides.map((trader) => (
+                        <ClientSlide
+                            key={trader.id}
```
Comment (sean-binary) [inline src/pages/home/_what-our-clients-say.js:241]: Good use of the `id` here 👍

**Comment by sean-binary** [comment]
Conflicts 🙏

File: src/components/custom/_dhero.js (hunk 1 above)
Comment (yashim-deriv) [reply src/components/custom/_dhero.js]: Updated this one and others as well 😄

File: src/components/layout/nav.js [hunk 5]
```diff
@@ -498,58 +491,17 @@ const NavDesktop = ({ base, is_ppc, is_ppc_redirect, is_logged_in }) => {
...
-                is_open={is_resources_open}
-                has_animation={has_resources_animation}
-                Content={() => <NavResources onClick={handleResourcesClick} />}
-                title={localize('Resources')}
-                description={localize(
-                    'Help yourself to various resources that can help you get the best out of your trading experience.',
-                )}
-            />
-
+        <div>
```
Comment (sean-binary) [inline src/components/layout/nav.js]: [Clarification] Can this be replaced with a react fragment or is a `div` required here?

Comment (yashim-deriv) [reply src/components/layout/nav.js]: 🤔  I think this one is from merging, i need to check

File: src/pages/careers/_who-we-look-for.js [hunk 6]
```diff
@@ -80,22 +80,12 @@ const ImageWrapper = styled.div`
     max-width: 65rem;
 `
 const query = graphql`
-    query {
+    {
```
Comment (sean-binary) [inline src/pages/careers/_who-we-look-for.js:83]: [Clarification] Any reason this is deleted?

File: src/pages/partners/_about-deriv.js [hunk 7]
```diff
@@ -7,14 +7,9 @@ import { localize } from 'components/localization'
 import device from 'themes/device'
 
 const query = graphql`
-    query {
+    {
```
Comment (sean-binary) [inline src/pages/partners/_about-deriv.js:10]: Same here.

File: src/pages/careers/_who-we-look-for.js (hunk 6 above)
Comment (sean-binary) [reply src/pages/careers/_who-we-look-for.js:83]: Note: This was part of the codemod migration as query wrappers are no longer required. They can all be deleted in the future.

**Review by sean-binary — DISMISSED** [review]

**Review by sean-binary — APPROVED** [review]

**Pull request merged**